internal/client/gandi.go

```diff
@@ -27,15 +28,18 @@ type Gandi struct {
 }
 
 // New instantiates a new Gandi client
-func New(apikey string, apiurl string, sharingID string, debug bool, dryRun bool, timeout time.Duration) *Gandi {
+func New(apikey string, token string, apiurl string, sharingID string, debug bool, dryRun bool, timeout time.Duration) *Gandi {
 	if apiurl == "" {
 		apiurl = config.APIURL
 	}
 	endpoint := apiurl + "/v5/"
 	if timeout == 0 {
 		timeout = config.Timeout
 	}
-	return &Gandi{apikey: apikey, endpoint: endpoint, sharingID: sharingID, debug: debug, dryRun: dryRun, timeout: timeout}
+	if apikey != "" {
+		log.Println("Warning: Authenticating using an API key is deprecated - please use a Personal Access Token instead.")
+	}
```

Comment:
could we add an `else` branch here testing for `token` and if not set exiting with an error? I guess at least one from apikey and token is required

Reply:
This gets a little more complex, and is an issue with the old API anyway. Currently the constructors don't have any error handling, and adding it would massively change the API downstream, and how it's used. I'll add a similar warning when no authentication is provided, but better handling can probably be done in a separate PR.